packages/stylist-core/src/parser.rs

```diff
@@ -20,46 +20,80 @@ use nom::{
 #[cfg(test)]
 use log::trace;
 
-/// A lightweight CSS Parser.
-#[derive(Debug)]
-pub(crate) struct Parser {}
-
+/// Wrap a parser, tracing input and output.
+// if not cfg(test), this would trip up clippy.
 #[allow(clippy::let_and_return)]
-impl Parser {
-    /// Returns Error when string is Empty
-    fn expect_non_empty(i: &str) -> std::result::Result<(), nom::Err<VerboseError<&str>>> {
+fn traced_context<I, O, F>(
+    ctx: &'static str,
+    mut p: impl nom::Parser<I, O, F>,
+) -> impl FnMut(I) -> IResult<I, O, F>
+where
+    I: fmt::Display + fmt::Debug + Clone,
+    O: fmt::Debug,
+    F: fmt::Debug + nom::error::ContextError<I>,
+{
+    move |i| {
+        #[cfg(test)]
+        trace!("> {}: {}", ctx, i);
```

Comment:
Better to give the tracing context a number?

Reply:
Meaning an indentation level, or number instead of `ctx: &str`, not sure I understand :)

Reply:
Sorry that I didn't explain this properly.

Sometimes it's possible to have the following situation:

```
> Rule:
... 
> Rule: 
...
> Rule:
...
> Rule:
...
> Rule:

...

< Rule:
... 
< Rule: 
...
< Rule:
...
< Rule:
...
< Rule:
```

So it might be handy to have a number to keep track of it:

```
> Rule 1:
... 
> Rule 2: 
...
> Rule 3:
...
> Rule 4:
...
> Rule 5:

...

< Rule 5:
... 
< Rule 4: 
...
< Rule 3:
...
< Rule 2:
...
< Rule 1:
```

I think indentation should work as well.